From dff669c576ce05b1890f4783271aa70137f7e5dc Mon Sep 17 00:00:00 2001
From: Dillibabu <dillibabug@ami.com>
Date: Fri, 7 Jul 2023 15:19:08 +0530
Subject: [PATCH] Board Mfg date  printf function modified.

Before it will print unspecified if time bytes are zero, now if value zero default year and date

will be printed.

Signed-off-by: Dillibabu <dillibabug@ami.com>
---
 lib/ipmi_fru.c | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/lib/ipmi_fru.c b/lib/ipmi_fru.c
index 4a5018d..7ab7802 100644
--- a/lib/ipmi_fru.c
+++ b/lib/ipmi_fru.c
@@ -1066,6 +1066,7 @@ fru_area_print_board(struct ipmi_intf * intf, struct fru_info * fru,
 	uint32_t fru_len;
 	uint32_t i;
 	time_t ts;
+	time_t tval;
 	uint8_t tmp[2];
 
 	fru_len = 0;
@@ -1100,8 +1101,11 @@ fru_area_print_board(struct ipmi_intf * intf, struct fru_info * fru,
 	 */
 	i = 3;
 
-	ts = ipmi_fru2time_t(&fru_data[i]);
-	printf(" Board Mfg Date        : %s\n", ipmi_timestamp_string(ts));
+	tval=((fru_data[i+2] << 16) + (fru_data[i+1] << 8) + (fru_data[i]));
+	tval=tval * 60;
+	tval=tval + 820454400;
+	printf(" Board Mfg Date        : %s", asctime(localtime(&tval)));
+
 	i += 3;  /* skip mfg. date time */
 
 	fru_area = get_fru_area_str(fru_data, &i);
